ui-lib/src/main/java/co/electriccoin/zcash/ui/screen/update_available/view/UpdateAvailableView.kt

```diff
@@ -0,0 +1,195 @@
+        }
+    }
+}
+
+@Composable
+private fun UpdateAvailableTopAppBar(updateInfo: UpdateInfo) {
+    SmallTopAppBar(
+        title = {
+            Text(
+                text = stringResource(
+                    updateInfo.isForce.let { force ->
+                        if (force)
+                            R.string.update_available_critical_header
+                        else
+                            R.string.update_available_header
+                    }
+                )
+            )
+        },
+    )
+}
+
+@Composable
+private fun UpdateAvailableBottomAppBar(
+    updateInfo: UpdateInfo,
+    onDownload: (state: UpdateState) -> Unit,
+    onLater: () -> Unit
+) {
+    Column {
+        PrimaryButton(
+            onClick = { onDownload(UpdateState.Running) },
+            text = stringResource(R.string.update_available_download_button),
+            modifier = Modifier
+                .fillMaxWidth()
+                .testTag(RestoreTag.BTN_DOWNLOAD),
+            enabled = updateInfo.state != UpdateState.Running
+        )
+
+        TertiaryButton(
+            onClick = onLater,
+            text = stringResource(
+                updateInfo.isForce.let { force ->
+                    if (force)
+                        R.string.update_available_later_disabled_button
+                    else
+                        R.string.update_available_later_enabled_button
+                }
+            ),
+            modifier = Modifier
+                .fillMaxWidth()
+                .testTag(RestoreTag.BTN_LATER),
+            enabled = !updateInfo.isForce && updateInfo.state != UpdateState.Running
+        )
+    }
+}
+
+@Composable
+private fun UpdateAvailableContentNormal(
+    onReference: () -> Unit
+) {
+    val context = LocalContext.current
+
+    Column(
+        Modifier.fillMaxWidth(),
+        horizontalAlignment = Alignment.CenterHorizontally,
+    ) {
+        // TODO [#17]: This suppression and magic number will get replaced once we have real assets
+        @Suppress("MagicNumber")
+        Image(
+            painter = painterResource(id = R.drawable.update_available_main_graphic),
+            contentDescription = stringResource(id = R.string.update_available_image_content_description),
+            Modifier.fillMaxSize(0.50f)
+        )
+
+        Body(
+            text = stringResource(id = R.string.update_available_description),
+            modifier = Modifier
+                .wrapContentHeight()
+                .align(Alignment.CenterHorizontally)
+        )
+
+        Reference(
+            text = stringResource(id = R.string.update_available_link_text),
+            tag = REFERENCE_TAG,
+            link = stringResource(id = R.string.update_available_link),
+            modifier = Modifier
+                .wrapContentHeight()
+                .align(Alignment.CenterHorizontally),
+            onClick = {
+                val storeIntent = PlayStoreUtil.newActivityIntent(context)
```

Comment:
Can we extract the Android-specific implementation here (which is using Context and startActivity) into a lambda that's passed into the view?  It would be similar in concept to how I injected copy to clipboard in the seed display screen during onboarding.

That helps us re-use the view even if we make the app multiplatform or support different non-Google Play app stores.

Reply:
Yes, sure, good idea!